src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1977,3 +1981,52 @@ def set_attention_slice(self, slice_size: Optional[int]):
 
         for module in modules:
             module.set_attention_slice(slice_size)
+
+    def load_workflow(self, workflow_id_or_path: Union[str, dict], filename: Optional[str] = None):
+        r"""Loads a workflow from the Hub or from a local path. Also patches the pipeline call arguments with values from the
+        workflow.
+
+        Args:
+            workflow_id_or_path (`str` or `dict`):
+                Can be either:
+
+                    - A string, the workflow id (for example `sayakpaul/sdxl-workflow`) of a workflow hosted on the
+                      Hub.
+                    - A path to a directory (for example `./my_workflow_directory`) containing the workflow file with
+                      [`Workflow.save_workflow`] or [`Workflow.push_to_hub`].
+                    - A Python dictionary.
+
+            filename (`str`, *optional*):
+                Optional name of the workflow file to load. Especially useful when working with multiple workflow
+                files.
+        """
+        filename = filename or WORKFLOW_NAME
+
+        # Load workflow.
+        if not isinstance(workflow_id_or_path, dict):
+            if os.path.isdir(workflow_id_or_path):
+                workflow_filepath = os.path.join(workflow_id_or_path, filename)
+            elif os.path.isfile(workflow_id_or_path):
+                workflow_filepath = workflow_id_or_path
+            else:
+                workflow_filepath = hf_hub_download(repo_id=workflow_id_or_path, filename=filename)
+            workflow = self._dict_from_json_file(workflow_filepath)
+        else:
+            workflow = workflow_id_or_path
+
+        # Handle generator.
+        seed = workflow.pop("seed", None)
+        if seed is not None:
+            generator = torch.manual_seed(seed)
+        else:
+            generator = None
+        workflow.update({"generator": generator})
```

Comment:
Can we serialize a `generator`?

Reply:
We're not serializing a generator. See here:
https://github.com/huggingface/diffusers/blob/feat/workflows/src/diffusers/workflow_utils.py#L62C1-L67C44

Reply:
Wondering if you'd also like to add info on the generator device.

Reply:
Done.